job/server/src/main/java/alluxio/worker/job/command/CommandHandlingExecutor.java

```diff
@@ -79,11 +81,23 @@ public CommandHandlingExecutor(JobServerContext jobServerContext,
     mTaskExecutorManager = Preconditions.checkNotNull(taskExecutorManager, "taskExecutorManager");
     mMasterClient = Preconditions.checkNotNull(masterClient, "masterClient");
     mWorkerNetAddress = Preconditions.checkNotNull(workerNetAddress, "workerNetAddress");
-    mHealthReporter = new JobWorkerHealthReporter();
+    if (ServerConfiguration.getBoolean(PropertyKey.TEST_MODE)) {
+      mHealthReporter = new AlwaysHealthyJobWorkerHealthReporter();
```

Comment:
Instead of doing a switch here, perhaps add logic in the `JobWorkerHealthReporter` to make the health factor configurable?

Reply:
I don't think I prefer that given that just mixes more logic inside JobWorkerHealthReporter file.

Reply:
Wouldn't the `JobWorkerHealthReporter` benefit from being able to be configured? Is `1.0 * # of available processors` always the right threshold?

Reply:
I think configuration may be necessary later but even then, I think a class that always returns True for isHealthy is likely better.

Reply:
I'm happy with a test version of the health reporter, but I don't like injecting it in the production code itself, it usually is handled though some kind of mocking/test framework.